driver/src/orderbook/streamed/state.rs

```diff
@@ -787,4 +787,106 @@ mod tests {
         assert_eq!(balance.get(&(address(2), 0)), Some(&12.into()));
         assert_eq!(balance.get(&(address(2), 1)), Some(&9.into()));
     }
+
+    #[test]
+    fn orderbook_solution_touching_same_order_twice() {
+        let mut state = state_with_fee();
```

Comment:
Could we not initialize the state as we want it without having to apply all the token listings and deposits?